src/offlineContentManager.ts

```diff
@@ -0,0 +1,211 @@
+import { SourceConfig } from './source';
+import {
+  BitmovinNativeOfflineEventData,
+  OfflineContentManagerListener,
+  OfflineEventType,
+} from './offlineContentManagerListener';
+import { OfflineDownloadRequest } from './offlineContentOptions';
+
+interface NativeOfflineModule extends NativeModule {
+  initWithConfig(
+    nativeId: string,
+    config: { offlineId: string; sourceConfig: SourceConfig }
+  ): Promise<void>;
+  getOfflineSourceConfig(nativeId: string): Promise<SourceConfig>;
+  getOptions(nativeId: string): Promise<void>;
+  process(nativeId: string, request: OfflineDownloadRequest): Promise<void>;
+  resume(nativeId: string): Promise<void>;
+  suspend(nativeId: string): Promise<void>;
+  cancelDownload(nativeId: string): Promise<void>;
+  deleteAll(nativeId: string): Promise<void>;
+  downloadLicense(nativeId: string): Promise<void>;
+  releaseLicense(nativeId: string): Promise<void>;
+  renewOfflineLicense(nativeId: string): Promise<void>;
+  release(nativeId: string): Promise<void>;
+}
+
+const OfflineModule =
+  NativeModules.BitmovinOfflineModule as NativeOfflineModule;
+
+/**
+ * Object used to configure a new `OfflineContentManager` instance.
+ */
+export interface OfflineContentConfig extends NativeInstanceConfig {
+  /**
+   * An identifier for this source that is unique within the location and must never change.
+   * The root folder will contain a folder based on this id.
+   */
+  offlineId: string;
+  /**
+   * The `SourceConfig` used to download the offline resources.
+   */
+  sourceConfig: SourceConfig;
+  /**
+   * The `OfflineContentManagerListener` where callbacks for event data will be passed to.
+   */
+  listener?: OfflineContentManagerListener;
+}
+
+/**
+ * Provides the means to download and store sources locally that can be played back with a Player
+ * without an active network connection.  An OfflineContentManager instance can be created via
+ * the constructor and will be idle until initialized.
+ */
+export class OfflineContentManager extends NativeInstance<OfflineContentConfig> {
+  isInitialized = false;
+  isDestroyed = false;
+  eventSubscription?: EmitterSubscription = undefined;
+
+  constructor(config: OfflineContentConfig) {
+    super(config);
+  }
+
+  /**
+   * Allocates the native `OfflineManager` instance and its resources natively.
+   * Registers the `DeviceEventEmitter` listener to receive data from the native `OfflineContentManagerListener` callbacks
+   */
+  initialize = (): Promise<void> => {
+    let initPromise = Promise.resolve();
+    if (!this.isInitialized && this.config) {
+      if (this.config.listener) {
+        this.eventSubscription = new NativeEventEmitter(
+          OfflineModule
+        ).addListener(
+          'BitmovinOfflineEvent',
+          (data?: BitmovinNativeOfflineEventData) => {
+            if (this.nativeId !== data?.nativeId) {
+              return;
+            }
+
+            if (data.eventType === OfflineEventType.onCompleted) {
```

Comment:
@rolandkakonyi I'm adding the missing event types.  I originally didn't include them because they were iOS only in the case of `onCanceled` and `onLicenseExpired`.  I opted to go with the android equivalent naming for `onLicenseRenewed`, which is `onDrmLicenseUpdated`.  I'm going to mark the functions in comment as being iOS only.  Is there a specific way you'd like these callbacks named?